### DISCUSSION TITLE
How to avoid unwanted completion

### DISCUSSION THREAD

**Discussion by danielRi** [nushell/nushell#5495] [Q&A]
I noticed that sometimes I get code completions where I´m not expecting them. I try to describe my problem here. Im not sure, maybe some people will say "this is expected behavior, and its good", but for me it seems it shouldnt be like this.

You can see an occasion of the unwanted completion of a folder name in the video. Here is what I´m doing in the video:

1. typing `cd aa`
2. hitting `tab`
3. there are no folders/files that start with `aa`, so the prompt correctly shows `NO RECORDS FOUND` - everything good so far
4. Now Im hitting backspace, deleting one a, changing the input to just `a`.

**Observed, unwanted behavior:**
The auto completion triggers, and I now have the whole folder name in my prompt input. I need to delete it completely, even though i just typed `aa`, which irritates me at this point (especially if I dont expect it and was hacking commands very quickly). I know I could delete the word quickly with hotkeys, but it´s still not preferred behavior.

**Expected:**

I would expected that I just have `a` in my input prompt, I can just delete it and then type the name from start again

https://user-images.githubusercontent.com/3048681/167540617-ea28013d-15f7-4674-85ed-98397b1af53b.mov

**Comment by merelymyself** [comment]
I don't think it's intended behaviour, because it doesn't make much sense to me. You should open an issue.

I'm not on the nushell design team though, so take my words with a pinch of salt.

**Comment by danielRi** [comment]
Hey @merelymyself Thanks for your reply. I created an issue. I think it´s worth let the devs have a look at this.
